include/rwqueue.h

```diff
@@ -131,6 +135,12 @@ class RWQueue {
 	// after stopping will not be queued.
 	bool BulkEnqueue(std::vector<T>& from_source, const size_t num_requested);
 
+	// Returns false and does nothing if queue is at capcity or queue is not running
+	// Otherwise, enqueues as many elements as possible until the queue is at capacity
+	// Moved elements will be erased from the source vector
+	// Elements not enqueued will be left in the source vector
+	bool NonblockingBulkEnqueue(std::vector<T>& from_source, const size_t num_requested);
```

Comment:
Wouldn't returning the number of queued elements be more useful than just true or false?

Reply:
Yeah that's probably better.  None of the current use-cases actually check the return value.  I was just trying to match existing API.

Reply:
Changed